--- a/tests/Makemodule.am
+++ b/tests/Makemodule.am
@@ -13,7 +13,7 @@
 
 CLEAN_LOCALS += clean-local-tests
 
-TS_OPTS = --nonroot
+TS_OPTS = --nonroot --verbose
 TS_COMMAND = $(top_srcdir)/tests/run.sh \
 	--srcdir=$(abs_top_srcdir) --builddir=$(abs_top_builddir) \
 	--parallel $(TS_OPTS)
--- a/tests/functions.sh
+++ b/tests/functions.sh
@@ -94,6 +94,15 @@
 		ts_report " $msg ($1)"
 	fi
 
+	if [ "$TS_VERBOSE" = "yes" ]; then
+		echo ========= script: $TS_SCRIPT =================
+		echo ================= OUTPUT =====================
+		cat $TS_OUTPUT
+		echo ================= EXPECTED ===================
+		cat $TS_EXPECTED
+		echo ==============================================
+	fi
+
 	return $ret
 }
 
